common/hyperscaler/rules/definitions.go

```diff
@@ -0,0 +1,162 @@
+package rules
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+)
+
+const (
+	PR_ATTR_NAME = "PR"
+	HR_ATTR_NAME = "HR"
+	EU_ATTR_NAME = "EU"
+	S_ATTR_NAME  = "S"
+
+	HYPERSCALER_LABEL = "hyperscalerType"
+	EUACCESS_LABEL    = "euAccess"
+	SHARED_LABEL      = "shared"
+
+	ASTERISK                 = "*"
+	ATTRIBUTE_WITH_VALUE     = "attr"
+	SIGNATURE_ATTR_SEPARATOR = ":"
+)
+
+var InputAttributes = []Attribute{
+	{
+		Name:            PR_ATTR_NAME,
+		Description:     "Platform Region",
+		Setter:          setPlatformRegion,
+		Getter:          func(r *Rule) string { return r.PlatformRegion },
+		MatchableGetter: func(r *MatchableAttributes) string { return r.PlatformRegion },
+		input:           true,
+		output:          false,
+		HasValue:        true,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			if labels[HYPERSCALER_LABEL] == "" {
+				labels[HYPERSCALER_LABEL] = r.Plan
+			} else {
+				labels[HYPERSCALER_LABEL] += "_" + r.PlatformRegion
+			}
+			return labels
+		},
+	},
+	{
+		Name:        HR_ATTR_NAME,
+		Description: "Hyperscaler Region",
+		Setter:      setHyperscalerRegion,
+		Getter:      func(r *Rule) string { return r.HyperscalerRegion },
+		input:       true,
+		output:      false,
+		HasValue:    true,
+
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			if labels[HYPERSCALER_LABEL] == "" {
+				labels[HYPERSCALER_LABEL] = r.Plan
+			} else {
+				labels[HYPERSCALER_LABEL] += "_" + r.HyperscalerRegion
+			}
+			return labels
+		},
+	},
+}
+
+var OutputAttributes = []Attribute{
+	{
+		Name:        EU_ATTR_NAME,
+		Description: "EU Access",
+		Setter:      setEuAccess,
+		Getter:      func(r *Rule) string { return strconv.FormatBool(r.EuAccess) },
+		input:       false,
+		output:      true,
+		HasValue:    false,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			if r.EuAccess {
+				labels[EUACCESS_LABEL] = "true"
+			}
+			return labels
+		},
+	},
+	{
+		Name:        S_ATTR_NAME,
+		Description: "Shared",
+		Setter:      setShared,
+		Getter:      func(r *Rule) string { return strconv.FormatBool(r.Shared) },
+		input:       false,
+		output:      true,
+		HasValue:    false,
+		ApplyLabel: func(r *Rule, labels map[string]string) map[string]string {
+			if r.Shared {
+				labels[SHARED_LABEL] = "true"
+			}
+
+			return labels
+		},
+	},
+}
+
+var AllAttributes = append(InputAttributes, OutputAttributes...)
+
+func setShared(r *Rule, value string) (*Rule, error) {
+	if r.Shared {
+		return nil, fmt.Errorf("Shared already set")
```

Comment:
Should we treat this as error and exit function?

Reply:
It is used to return correct error during parsing.

Reply:
The question is why setting shared twice (more than once) is an error?
Why this r.Shared is write once only?